src/msg/simple/Accepter.cc

```diff
@@ -313,7 +324,21 @@ void *Accepter::entry()
       }
       ldout(msgr->cct,1) << __func__ << " poll got error"  
  			  << " errno " << errno << " " << cpp_strerror(errno) << dendl;
-      break;
+ 			  
+ 	  if (msgr->cct->get_module_type() == CEPH_ENTITY_TYPE_OSD && hb){
```

Comment:
Is there a reason to think this issue only applies to the OSD?

I think we'd be better off with some setup state that tells the Accepter to make use of a heartbeat map or not. Perhaps just allow daemons to set a heartbeat_map pointer which is default-initialized to NULL, and if it's set the Accepter uses it? :)

Reply:
Yes , it's better to  use a better format of  heartbeat map for accepter  worker.
I think this issue can influence any other daemon, but osd daemons communication failure is important than other damons. There is no other osd if the osd can not connect to.

Reply:
The reason of only applying to the OSD is currently heartbeat is mainly for checking timeout of work threads,
so IIUC, monitor daemon won't check heartbeat. Adding osd accepter to heartbeat is tricky simple solution but it will not work perperly for other deamons.